index.bs

```diff
@@ -571,67 +573,110 @@ When this method is invoked, the user agent MUST execute the following algorithm
-    |clientDataHash|.
+1. If the {{AssertionOptions/extensions}} member of {{options}} is [=present=], then [=map/for each=]
+    |extension| → |argument| of <code>{{options}}.{{AssertionOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, then [=continue=].
+    1. Otherwise, let |result| be the result of running |extension|'s [=client processing=] algorithm on |argument|. If the
+        algorithm returned an error, [=continue=].
+    1. [=list/Append=] |result| to |clientExtensions|.
+
+1. Let |clientData| be a new {{ClientData}} instance whose fields are:
+    : {{challenge}}
+    :: The [=base64url encoding=] of {{assertionChallenge}}
+    : {{origin}}
+    :: The [=unicode serialization of an origin|unicode serialization=] of |rpId|
+    : {{hashAlg}}
+    :: The [=recognized algorithm name=] of the hash algorithm selected by the client for generating the
+        [=hash of the serialized client data=]
+    : {{tokenBinding}}
+    :: The [=Token Binding ID=] associated with |callerOrigin| (if any)
+    : {{ClientData/extensions}}
+    :: |clientExtensions|
+
+1. Let |clientDataJSON| be the [=JSON-serialized client data=] constructed from |clientData|.
+
+1. Let |clientDataHash| be the [=hash of the serialized client data=] represented by |clientDataJSON|.
+
+1. Let |issuedRequests| be a new [=ordered set=]. 
+
+1. For each |authenticator| currently available on this platform, perform the following steps:
+
+    1. Let |credentialList| be a new [=list=].
 
-1. Initialize |issuedRequests| to an empty list.
-
-1. For each authenticator currently available on this platform, perform the following steps:
-    - If the {{AssertionOptions/allowList}} member of {{options}} is empty, let |credentialList| be an empty list. Otherwise,
-        execute a platform-specific procedure to determine which, if any, credentials listed in {{AssertionOptions/allowList}}
-        might be present on this authenticator, and set |credentialList| to this filtered list. If no such filtering is
-        possible, set |credentialList| to an empty list.
-    - For each credential C within the |credentialList| that has a non-empty {{transports}} list, optionally use only the
-        specified transports to get assertions using credential C.
-    - If the above filtering process concludes that none of the credentials on the {{AssertionOptions/allowList}} can possibly
-        be on this authenticator, do not perform any of the following steps for this authenticator, and proceed to the next
-        authenticator (if any).
-    - Asynchronously invoke the [=authenticatorGetAssertion=] operation on this authenticator with |rpId|,
-        |clientDataHash|, |credentialList|, and |clientExtensions| as parameters.
-    - Add an entry to |issuedRequests|, corresponding to this request.
+    1. [=list/For each=] credential |C| in <code>{{options}}.{{AssertionOptions/allowList}}</code> that is present on this 
```

Comment:
How can the client tell if a credential is on a particular authenticator? This is only possible in certain circumstances. Maybe we should flip this. Something like "For each credential in allowList, append to credentialList. Some clients may be able to determine, by client-specific methods, the presence or absence of credentials on certain authenticators. In this case, the client MAY skip appending those credentials which it knows to not be on the authenticator."

Thoughts?